infra/baseServer_test.go

```diff
@@ -160,10 +177,114 @@ func TestAgentAgentMessage(t *testing.T) {
 	}
 }
 
-func TestAgentListeningSpinner(t *testing.T) {
+func TestAgentListeningSpinnerOpen(t *testing.T) {
+	const numAgents = 3
+	m := make([]infra.AgentGeneratorCountPair[ITestBaseAgent], 1)
+	m[0] = infra.MakeAgentGeneratorCountPair(NewTestAgent, numAgents)
+	server := infra.CreateServer[ITestBaseAgent](m, 1, time.Second, 2)
+	server.Initialise()
+	msg := NewTestMessage()
+	agentMap := server.GetAgentMap()
+	arrayOfIDs := make([]uuid.UUID, numAgents)
+	i := 0
+	for id := range agentMap {
+		arrayOfIDs[i] = id
+		i++
+	}
+	server.BeginAgentListeningSession()
+	server.SendMessage(msg, arrayOfIDs)
+	fmt.Println(arrayOfIDs)
+	//time.Sleep(10*time.Second)
+	server.EndAgentListeningSession()
+	for _, ag := range agentMap {
+		testAgent := ag.(*TestAgent)
+		if testAgent.receivedMessage == false {
+			t.Errorf("agent did not receive a message")
+		}
+	}
```

Comment:
it's worth logging the id here as well, for debugging:

```suggestion
	for id, ag := range agentMap {
		testAgent := ag.(*TestAgent)
		if testAgent.receivedMessage == false {
			t.Errorf("agent %s did not receive a message", id)
		}
	}
```